cmd/gen-go-sample/inittree.go

```diff
@@ -222,32 +221,58 @@ func (t *initTree) print(w *bufio.Writer, g *generator, ind int) error {
 	}
 	g.imports[impSpec] = true
 
+	// map field name to oneof name
+	var oneofs map[string]string
```

Comment:
Does this PR only deals with writing oneof fields but not parsing a tree with oneof fields?

Reply:
Parsing oneof fields already work. Oneof fields look like normal fields, they just have the oneof_index member set. We look for that in L230